cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1004,7 +1013,7 @@ protected Future<Void> kafkaRollToAddOrRemoveVolumesInStatefulSet(JbodStorage jb
         return podOperator.listAsync(reconciliation.namespace(), kafka.getSelectorLabels())
                 .compose(pods -> {
                     for (Pod pod : pods) {
-                        if (!needsRestartBecauseAddedOrRemovedJbodVolumes(pod, jbodStorage, currentReplicas, kafka.getReplicas()).isEmpty())   {
+                        if (needsRestartBecauseAddedOrRemovedJbodVolumes(pod, jbodStorage, currentReplicas, kafka.getReplicas()).shouldRoll())   {
```

Comment:
@scholzj what do you think of this? Encapsulates the logic of "if empty, then don't roll"

Reply:
That looks definitely more readable.

Reply:
Thanks :)  as I said an earlier comment,  I prefer the null vs. empty logic also, but adding that would take some time to ensure null safety, so I'll leave that for probably the ZK restart reasons PR.